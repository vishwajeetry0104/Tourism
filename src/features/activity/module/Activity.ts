import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
import type {PayloadAction} from '@reduxjs/toolkit';
import API from '../../../api/API';

type Activity = {
  name: string
}

type ActivityDetail = {
  name: string,
  description: string,
  image: string,
  activities: Array<Activity>
}

interface InitialStateType {
  loading: boolean;
  activityTypes: {
    'Surfing': ActivityDetail,
    'Traditional Festivals': ActivityDetail,
    'Volcanoes': ActivityDetail
  };
  hasFailed: boolean;
}

const defaultActivityDetail = {
  name: '',
  description: '',
  image: '',
  activities: []
};

export const initialState: InitialStateType = {
  loading: false,
  activityTypes: {
    'Surfing': defaultActivityDetail,
    'Traditional Festivals': defaultActivityDetail,
    'Volcanoes': defaultActivityDetail
  },
  hasFailed: false
};

export const fetchActivityDetail = createAsyncThunk('activity/details', async ({activityType}: {activityType: string}) => {
  const response = await API.get(`/v1/activities/${activityType}`);
  return response?.data;
});

export const activitySlice = createSlice({
  name: 'activity',
  initialState,
  reducers: {
    resetActivity: (state: InitialStateType) => {
      state.activityTypes = {
        'Surfing': defaultActivityDetail,
        'Traditional Festivals': defaultActivityDetail,
        'Volcanoes': defaultActivityDetail
      };
      state.loading = false;
      state.hasFailed = false;
    },
  },
  extraReducers: builder => {
    builder.addCase(
      fetchActivityDetail.fulfilled,
      (state: InitialStateType, action: PayloadAction<ActivityDetail, any, {arg: {activityType: string}}>) => {
        const {activityType} = action?.meta?.arg ?? {activityType: ''};
        state.activityTypes[activityType] = action.payload;
        state.loading = false;
        state.hasFailed = false;
      },
    );
    builder.addCase(fetchActivityDetail.pending, (state: InitialStateType) => {
      state.loading = true;
      state.hasFailed = false;
    });
    builder.addCase(fetchActivityDetail.rejected, (state: InitialStateType) => {
      state.loading = false;
      state.hasFailed = true;
    });
  },
});

export const {resetActivity} = activitySlice.actions;

export default activitySlice.reducer;
