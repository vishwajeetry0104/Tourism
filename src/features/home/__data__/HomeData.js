const highLightData = [
  {
    title: 'Surfing',
    description: 'Best Hawaiian islands for surfing.',
    image: 'https://storage.googleapis.com/topics-images/web-dev-images/surfing_low.png'
  },
  {
    title: 'Traditional Festivals',
    description: 'Best places to attend traditional festivals in Hawaii',
    image: 'https://storage.googleapis.com/topics-images/web-dev-images/traditional_festivals_low.png'
  },
  {
    title: 'Volcanoes',
    description: 'Volcanic conditions can change at any time.',
    image: 'https://storage.googleapis.com/topics-images/web-dev-images/volcanoes_low.png'
  }
];

const categoryData = [
  {name:'Adventure'},
  {name:':Culinary'},
  {name:':Eco-tourism'},
  {name:':Family'},
  {name:':Sport'}
];

export default {
  categoryData,
  highLightData
}
