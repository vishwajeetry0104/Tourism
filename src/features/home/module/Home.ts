import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
import type {PayloadAction} from '@reduxjs/toolkit';
import API from '../../../api/API';

interface InitialStateType {
  loading: boolean;
  data: [];
  hasFailed: boolean;
}

export const initialState: InitialStateType = {
  loading: false,
  data: [],
  hasFailed: false,
};

export const fetchHighlights = createAsyncThunk('home/highlights', async () => {
  const response = await API.get('/highlights');
  return response?.data;
});

export const homeSlice = createSlice({
  name: 'home',
  initialState,
  reducers: {
    resetHome: (state: InitialStateType) => {
      state.data = [];
      state.loading = false;
      state.hasFailed = false;
    },
  },
  extraReducers: builder => {
    builder.addCase(
      fetchHighlights.fulfilled,
      (state: InitialStateType, action: PayloadAction<any>) => {
        state.loading = false;
        state.hasFailed = false;
      },
    );
    builder.addCase(fetchHighlights.pending, (state: InitialStateType) => {
      state.loading = true;
      state.hasFailed = false;
    });
    builder.addCase(fetchHighlights.rejected, (state: InitialStateType) => {
      state.loading = false;
      state.hasFailed = true;
    });
  },
});

export const {resetHome} = homeSlice.actions;

export default homeSlice.reducer;
