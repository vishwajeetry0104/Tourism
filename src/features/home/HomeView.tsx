import React, {useEffect} from 'react';
import {useAppDispatch, useAppSelector} from "../../store/Hooks";
import {RootState} from "../../store/Store";
import HighlightCard from "../../components/highlightCard/HighlightCard";
import { ImageBackground, useWindowDimensions, Text, View, ScrollView, FlatList, Alert } from "react-native";
import HuwaiLandscape from '../../assets/images/huwai_landscape.png';
import CategoryItem from "../../components/categoryItem/CategoryItem";
import TappableButton from "../../components/button/TappableButton";
import TravelGuideCard from "../../components/travelGuide/TravelGuideCard";
import {fetchHighlights, resetHome, fetchCategories} from "./module/Home";
import Spinner from "../../components/spinner/Spinner";
import InfoCard from "../../components/infoCard/InfoCard";
import Styles from './HomeStyles';

import GradientText from "../../components/gradientText/GradientText";

const HomeView = ({navigation}) => {
  const {width} = useWindowDimensions();
  const dispatch = useAppDispatch();

  const {
    highLights,
    loadingHighlights,
    hasHighlightsFailed,
    hasCategoriesFailed,
    categories,
    loadingCategories
  } = useAppSelector((state: RootState) => state.home);

  useEffect(() => {
    dispatch(fetchHighlights());
    dispatch(fetchCategories());
    return () => {
      dispatch(resetHome());
    }
  }, []);

  const getRouteName = (title: string) => {
    switch (title) {
      case 'Surfing': return 'Surfing';
      case 'Traditional Festivals': return 'Hula';
      case 'Volcanoes': return 'Vulcano';
      default: return '';
    }
  }

  const renderItem = ({item}) => {
    const onHighlightPress = () => {
      const routeName = getRouteName(item.title);
      navigation.navigate(routeName);
    }
    return (
      <HighlightCard
        testID={item.title}
        title={item.title}
        description={item.description}
        image={item.image}
        onHighlightPress={onHighlightPress}
      />
    )
  }

  const listEmptyComponent = () => {
    if(hasHighlightsFailed) {
      return (
        <View style={[{width: width}, Styles.flatListEmptyComponentContainer]}>
          <InfoCard
            testID='infoCard_highlights'
            headingTitle='No results'
            message={hasHighlightsFailed ? 'Something went wrong': 'Nothing to display'}
          />
        </View>
      )
    }
    return (
      <View style={[{width: width}, Styles.flatListEmptyComponentContainer]}>
        <Spinner testID="loader" animating={loadingHighlights} subTitle="highlights" inline />
      </View>
    )
  };

  const onPressCategory = (name) => {
    Alert.alert(`${name} pressed`);
  }

  return (
    <>
      <ScrollView style={Styles.scrollViewContainer} contentContainerStyle={Styles.contentContainerStyle}>
        <ImageBackground source={HuwaiLandscape} style={{height: (width / 3) * 4, width: width}} resizeMode="cover">
          <GradientText colors={['rgba(255, 255, 255, 0.05)', 'rgba(255, 255, 255, 0.90)']} style={Styles.imageBackgroundText}>
            Welcome to Hawaii
          </GradientText>
        </ImageBackground>
        <Text style={Styles.sectionLabelText}>Highlights</Text>
        <FlatList
          testID="FlatList"
          keyExtractor={(item, index) => index.toString()}
          data={highLights}
          renderItem={renderItem}
          ListEmptyComponent={listEmptyComponent()}
          horizontal
          contentContainerStyle={Styles.flatListContentContainerStyle}
          extraData={{
            hasHighlightsFailed,
            loadingHighlights,
            highLights
          }}
        />
        <Text style={[Styles.sectionLabelText, {backgroundColor: 'transparent'}]}>Catergories</Text>
        {hasCategoriesFailed ? (
          <View style={[Styles.flatListEmptyComponentContainer, {backgroundColor: 'transparent'}]}>
            <InfoCard
              testID='infoCard_categories'
              headingTitle='No results'
              message={hasCategoriesFailed ? 'Something went wrong': 'Nothing to display'}
            />
          </View>
        ) : categories.length === 0 && loadingCategories && (
          <View style={[Styles.flatListEmptyComponentContainer, {backgroundColor: 'transparent'}]}>
            <Spinner testID="loader" animating={loadingHighlights} subTitle="Categories" inline />
          </View>
        )}
        {categories.map((item) => {
          return (
            <CategoryItem key={item.name} name={item.name} onPressCategory={() => onPressCategory(item.name)} />
          )
        })}
        <TravelGuideCard name="Hadwin Malone" description="Guide since 2012" onContactPress={() => {}} />
      </ScrollView>
      <View style={Styles.floatingButtonContainer}>
        <TappableButton title="Book a trip" onButtonPress={() => {}} />
      </View>
    </>
  );
};

export default HomeView;
