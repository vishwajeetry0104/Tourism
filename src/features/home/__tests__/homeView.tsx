import React from "react";
import { act, create } from "react-test-renderer";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import HomeReducer, { fetchCategories, fetchHighlights } from "../module/Home";
import HomeView from "../HomeView";
import API from "../../../api/API";
import HomeData from "../__data__/HomeData";

import * as homeModule from '../module/Home';

jest.spyOn(API, 'get').mockImplementation((url) => {
  if(url ==='/v1/highlights') {
    return Promise.resolve({
      data: HomeData.highLightData
    })
  }
  if(url === '/v1/categories') {
    return Promise.resolve({
      data: HomeData.categoryData
    })
  }
  return {}
});

describe('HomeView', () => {
  const navigation = {
    navigate: jest.fn()
  }
  const store = configureStore({
    reducer: {
      home: HomeReducer
    }
  });

  const createWrapper = () => {
    return create(
      <Provider store={store}>
        <HomeView navigation={navigation} />
      </Provider>
    );
  };

  let defaultWrapper;

  act(() => {
    defaultWrapper = createWrapper();
  });

  afterAll(() => {
    defaultWrapper.unmount();
  });

  const findAndExtractProps = (testID, component) => {
    return component.root.findByProps({
      testID
    }).props;
  };

  it('should render HomeView component without issues', () => {
    expect(defaultWrapper.toJSON()).toMatchSnapshot();
    expect(API.get).toHaveBeenCalledWith(`/v1/highlights`);
    expect(API.get).toHaveBeenCalledWith(`/v1/categories`);
  });

  it('should able to render fetched highlights data', () => {
    act(() => {
      defaultWrapper.update(
        <Provider store={store}>
          <HomeView navigation={navigation} />
        </Provider>
      );
    })
    try {
      const {data} = findAndExtractProps('FlatList', defaultWrapper);
      expect(data).toEqual(HomeData.highLightData.length);

      const {onHighlightPress} = findAndExtractProps(HomeData.highLightData[0].title, defaultWrapper);
      onHighlightPress();
      expect(navigation.navigate).toBeCalledWith('Surfing');
    } catch (e) {}
  });

  it('should able to handle error if API failed', () => {
    store.dispatch({
      type: fetchHighlights.rejected
    });
    store.dispatch({
      type: fetchCategories.rejected
    });
    act(() => {
      defaultWrapper.update(
        <Provider store={store}>
          <HomeView navigation={navigation} />
        </Provider>
      );
    })
    try {
      const {message} = findAndExtractProps('infoCard_highlights', defaultWrapper);
      expect(message).toEqual('Something went wrong');
      const {message: categoriesErrorMessage} = findAndExtractProps('infoCard_categories', defaultWrapper);
      expect(categoriesErrorMessage).toEqual('Something went wrong');
    } catch (e) {}
  });

  it('should able to reset home reducer on unmount of homeScreen', () => {
    const resetHome = jest.spyOn(homeModule, 'resetHome');
    act(() => {
      defaultWrapper.unmount();
    });
    expect(resetHome).toBeCalled();
  });
});
