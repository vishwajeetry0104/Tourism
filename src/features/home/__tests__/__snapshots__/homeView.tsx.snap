// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView should render HomeView component without issues 1`] = `
[
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 100,
      }
    }
    style={
      {
        "backgroundColor": "#E6F2F2",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          {
            "height": 1000,
            "width": 750,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            {
              "process": [Function],
            }
          }
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": 1000,
                "width": 750,
              },
              undefined,
            ]
          }
        />
        <Text
          style={
            [
              {
                "color": "#FFF",
                "fontSize": 56,
                "fontWeight": "700",
                "lineHeight": 56,
                "marginTop": "40%",
                "textAlign": "center",
              },
              {
                "opacity": 0,
              },
            ]
          }
        >
          Welcome to Hawaii
        </Text>
      </View>
      <Text
        style={
          {
            "backgroundColor": "#FFF",
            "color": "#001A1A",
            "fontSize": 16,
            "fontWeight": "700",
            "lineHeight": 20,
            "padding": 16,
          }
        }
      >
        Highlights
      </Text>
      <RCTScrollView
        ListEmptyComponent={
          <View
            style={
              [
                {
                  "width": 750,
                },
                {
                  "backgroundColor": "#FFF",
                  "height": 200,
                },
              ]
            }
          >
            <Spinner
              animating={false}
              inline={true}
              subTitle="highlights"
              testID="loader"
            />
          </View>
        }
        contentContainerStyle={
          {
            "alignItems": "flex-start",
            "backgroundColor": "#FFFFFF",
          }
        }
        data={
          [
            {
              "description": "Best Hawaiian islands for surfing.",
              "image": "https://storage.googleapis.com/topics-images/web-dev-images/surfing_low.png",
              "title": "Surfing",
            },
            {
              "description": "Best places to attend traditional festivals in Hawaii",
              "image": "https://storage.googleapis.com/topics-images/web-dev-images/traditional_festivals_low.png",
              "title": "Traditional Festivals",
            },
            {
              "description": "Volcanic conditions can change at any time.",
              "image": "https://storage.googleapis.com/topics-images/web-dev-images/volcanoes_low.png",
              "title": "Volcanoes",
            },
          ]
        }
        extraData={
          {
            "hasHighlightsFailed": false,
            "highLights": [
              {
                "description": "Best Hawaiian islands for surfing.",
                "image": "https://storage.googleapis.com/topics-images/web-dev-images/surfing_low.png",
                "title": "Surfing",
              },
              {
                "description": "Best places to attend traditional festivals in Hawaii",
                "image": "https://storage.googleapis.com/topics-images/web-dev-images/traditional_festivals_low.png",
                "title": "Traditional Festivals",
              },
              {
                "description": "Volcanic conditions can change at any time.",
                "image": "https://storage.googleapis.com/topics-images/web-dev-images/volcanoes_low.png",
                "title": "Volcanoes",
              },
            ],
            "loadingHighlights": false,
          }
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        testID="FlatList"
        viewabilityConfigCallbackPairs={[]}
      >
        <View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 8,
                  "elevation": 7,
                  "justifyContent": "center",
                  "margin": 16,
                  "shadowColor": "rgba(0, 128, 128, 0.16)",
                  "shadowOffset": {
                    "height": 5,
                    "width": 2,
                  },
                  "shadowOpacity": 0.29,
                  "shadowRadius": 4.65,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://storage.googleapis.com/topics-images/web-dev-images/surfing_low.png",
                  }
                }
                style={
                  {
                    "borderTopLeftRadius": 8,
                    "borderTopRightRadius": 8,
                    "height": 170,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "padding": 24,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#008080",
                      "fontSize": 24,
                      "fontWeight": "700",
                      "paddingBottom": 16,
                    }
                  }
                >
                  Surfing
                </Text>
                <Text
                  style={
                    {
                      "color": "#001A1A",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    }
                  }
                >
                  Best Hawaiian islands for surfing.
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "flex-end",
                      "height": 40,
                      "paddingTop": 6,
                      "width": "100%",
                    }
                  }
                >
                  <SvgMock
                    height={40}
                    width={40}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 8,
                  "elevation": 7,
                  "justifyContent": "center",
                  "margin": 16,
                  "shadowColor": "rgba(0, 128, 128, 0.16)",
                  "shadowOffset": {
                    "height": 5,
                    "width": 2,
                  },
                  "shadowOpacity": 0.29,
                  "shadowRadius": 4.65,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://storage.googleapis.com/topics-images/web-dev-images/traditional_festivals_low.png",
                  }
                }
                style={
                  {
                    "borderTopLeftRadius": 8,
                    "borderTopRightRadius": 8,
                    "height": 170,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "padding": 24,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#008080",
                      "fontSize": 24,
                      "fontWeight": "700",
                      "paddingBottom": 16,
                    }
                  }
                >
                  Traditional Festivals
                </Text>
                <Text
                  style={
                    {
                      "color": "#001A1A",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    }
                  }
                >
                  Best places to attend traditional festivals in Hawaii
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "flex-end",
                      "height": 40,
                      "paddingTop": 6,
                      "width": "100%",
                    }
                  }
                >
                  <SvgMock
                    height={40}
                    width={40}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onFocusCapture={[Function]}
            onLayout={[Function]}
            style={
              [
                {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 8,
                  "elevation": 7,
                  "justifyContent": "center",
                  "margin": 16,
                  "shadowColor": "rgba(0, 128, 128, 0.16)",
                  "shadowOffset": {
                    "height": 5,
                    "width": 2,
                  },
                  "shadowOpacity": 0.29,
                  "shadowRadius": 4.65,
                  "width": 600,
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://storage.googleapis.com/topics-images/web-dev-images/volcanoes_low.png",
                  }
                }
                style={
                  {
                    "borderTopLeftRadius": 8,
                    "borderTopRightRadius": 8,
                    "height": 170,
                    "width": "100%",
                  }
                }
              />
              <View
                style={
                  {
                    "padding": 24,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#008080",
                      "fontSize": 24,
                      "fontWeight": "700",
                      "paddingBottom": 16,
                    }
                  }
                >
                  Volcanoes
                </Text>
                <Text
                  style={
                    {
                      "color": "#001A1A",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 20,
                      "paddingBottom": 8,
                    }
                  }
                >
                  Volcanic conditions can change at any time.
                </Text>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "flex-end",
                      "height": 40,
                      "paddingTop": 6,
                      "width": "100%",
                    }
                  }
                >
                  <SvgMock
                    height={40}
                    width={40}
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <Text
        style={
          [
            {
              "backgroundColor": "#FFF",
              "color": "#001A1A",
              "fontSize": 16,
              "fontWeight": "700",
              "lineHeight": 20,
              "padding": 16,
            },
            {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        Catergories
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 4,
            "padding": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#001A1A",
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "400",
              "paddingRight": 8,
              "width": "80%",
            }
          }
        >
          Adventure
        </Text>
        <View
          style={{}}
        >
          <SvgMock
            height={16}
            width={16}
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 4,
            "padding": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#001A1A",
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "400",
              "paddingRight": 8,
              "width": "80%",
            }
          }
        >
          :Culinary
        </Text>
        <View
          style={{}}
        >
          <SvgMock
            height={16}
            width={16}
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 4,
            "padding": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#001A1A",
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "400",
              "paddingRight": 8,
              "width": "80%",
            }
          }
        >
          :Eco-tourism
        </Text>
        <View
          style={{}}
        >
          <SvgMock
            height={16}
            width={16}
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 4,
            "padding": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#001A1A",
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "400",
              "paddingRight": 8,
              "width": "80%",
            }
          }
        >
          :Family
        </Text>
        <View
          style={{}}
        >
          <SvgMock
            height={16}
            width={16}
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 16,
            "marginVertical": 4,
            "padding": 24,
          }
        }
      >
        <Text
          style={
            {
              "color": "#001A1A",
              "flexWrap": "wrap",
              "fontSize": 16,
              "fontWeight": "400",
              "paddingRight": 8,
              "width": "80%",
            }
          }
        >
          :Sport
        </Text>
        <View
          style={{}}
        >
          <SvgMock
            height={16}
            width={16}
          />
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flex": 1,
            "justifyContent": "center",
            "margin": 16,
            "padding": 24,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View>
            <Text
              style={
                {
                  "color": "#001A1A",
                  "fontSize": 24,
                  "fontWeight": "700",
                  "paddingBottom": 16,
                }
              }
            >
              Hadwin Malone
            </Text>
            <Text
              style={
                {
                  "color": "#001A1A",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "paddingBottom": 8,
                }
              }
            >
              Guide since 2012
            </Text>
          </View>
          <SvgMock
            height={74}
            width={74}
          />
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "flex-start",
              "backgroundColor": "#FFF",
              "borderColor": "#008080",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginTop": 16,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#008080",
                "fontSize": 16,
                "fontWeight": "700",
                "paddingHorizontal": 24,
                "paddingVertical": 8,
              }
            }
          >
            Contact
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    style={
      {
        "bottom": 0,
        "left": 0,
        "paddingBottom": 16,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#008080",
          "borderRadius": 8,
          "elevation": 7,
          "marginHorizontal": 16,
          "paddingBottom": 11,
          "paddingHorizontal": 24,
          "paddingTop": 9,
          "shadowColor": "rgba(7, 56, 56, 0.80)",
          "shadowOffset": {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.29,
          "shadowRadius": 4.65,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "700",
            "lineHeight": 20,
          }
        }
      >
        Book a trip
      </Text>
    </View>
  </View>,
]
`;
